services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDbDocumentBuilder.java

```diff
@@ -142,10 +143,18 @@ public MongoDbDocumentBuilder withAuthId(final String authId) {
     public MongoDbDocumentBuilder withDeviceFilters(final List<Filter> filters) {
 
         filters.forEach(filter -> {
-            // TODO: To implement when filter values contain patterns such as * or %
-            document.put(mapDeviceField(filter.getField()), filter.getValue());
+            if (filter.getValue() instanceof String) {
+                final String value = (String) filter.getValue();
+                if (value.contains("*") || value.contains("?")) {
```

Comment:
does it make a big difference to handle the two cases differently? I would assume that MongoDB internally treats both cases the same anyway, i.e. wrapping a literal string with `^` and `$`, thus turning it into a regex.
We could then simply let *DeviceRegistryUtils.getRegexExpressionForSearchOperation* handle all filters ...

Reply:
I didn't find any official info on how it is being handled internally by MongoDB. Hence I used the above approach. If the value doesn't contain '*' or '?', then the query wouldn't involve _regex_ keyword. Based on the above approach it would be 
`{"device-id" : "4711"}}`. Else it would be`{"field",  {"device-id" : {"regex" : "^4711$"}}`. If you think it doesn't make any difference, I can treat both the cases same.

Reply:
we can refactor it if we run into another place where we need to transform wildcards into a regex ...